domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/transfers/TransferTest.java

```diff
@@ -0,0 +1,145 @@
+        //noinspection unchecked to avoid lint
+        final List<MissingRequiredValueNotice> noticeCollection =
+                (List<MissingRequiredValueNotice>) entityBuildResult.getData();
+        final MissingRequiredValueNotice notice = noticeCollection.get(0);
+
+        assertEquals("transfers.txt", notice.getFilename());
+        assertEquals("from_stop_id", notice.getFieldName());
+        assertEquals("null;stop id 1", notice.getEntityId());
+        assertEquals(1, noticeCollection.size());
+    }
+
+    // Field toStopId is annotated as `@NonNull` but test require this field to be null. Therefore annotation
+    // "@SuppressWarnings("ConstantConditions")" is used here to suppress lint.
+    @Test
+    void createTransferWithNullToStopIdShouldGenerateNotice() {
+        final Transfer.TransferBuilder underTest = new Transfer.TransferBuilder();
+
+        //noinspection ConstantConditions
+        final EntityBuildResult<?> entityBuildResult = underTest.fromStopId(FROM_STOP_ID)
+                .toStopId(null)
+                .transferType(VALID_TRANSFER_TYPE_VALUE)
+                .minTransferTime(VALID_MIN_TRANSFER_TIME_VALUE)
+                .build();
+
+        assertTrue(entityBuildResult.getData() instanceof List);
+        //noinspection unchecked to avoid lint
+        final List<MissingRequiredValueNotice> noticeCollection =
+                (List<MissingRequiredValueNotice>) entityBuildResult.getData();
+        final MissingRequiredValueNotice notice = noticeCollection.get(0);
+
+        assertEquals("transfers.txt", notice.getFilename());
+        assertEquals("to_stop_id", notice.getFieldName());
+        assertEquals("stop id 0;null", notice.getEntityId());
+        assertEquals(1, noticeCollection.size());
+    }
+
+    @Test
+    void createTransferWithUnexpectedTransferTypeValueShouldGenerateNotice() {
+        final Transfer.TransferBuilder underTest = new Transfer.TransferBuilder();
+
+        final EntityBuildResult<?> entityBuildResult = underTest.fromStopId(FROM_STOP_ID)
+                .toStopId(TO_STOP_ID)
+                .transferType(55)
+                .minTransferTime(VALID_MIN_TRANSFER_TIME_VALUE)
+                .build();
+
+        assertTrue(entityBuildResult.getData() instanceof List);
+        //noinspection unchecked to avoid lint
+        final List<UnexpectedEnumValueNotice> noticeCollection =
+                (List<UnexpectedEnumValueNotice>) entityBuildResult.getData();
+        final UnexpectedEnumValueNotice notice = noticeCollection.get(0);
+
+        assertEquals("transfers.txt", notice.getFilename());
+        assertEquals("transfer_type", notice.getFieldName());
+        assertEquals("stop id 0;stop id 1", notice.getEntityId());
+        assertEquals("55", notice.getEnumValue());
+        assertEquals(1, noticeCollection.size());
+    }
+
+    @Test
+    void createTransferWithInvalidMinTransferTimeValueShouldGenerateNotice() {
```

Comment:
As discussed, range validation is already performed at type validation stage. This is more of a sucpicious values validation. Thresholds should come from configuration file and lead to emission of warning notices.

Reply:
Related issue: https://github.com/MobilityData/gtfs-validator/issues/159#issuecomment-626788425

Reply:
It seems that we might have neglected the fact that Transfer is a domain entity: it has no access to the user configurable thresholds contained in the ExecParamRepo content (adapter layer). 

In this particular case, when building the Transfer entity, we already checked that  minTransferTime >0). Hence, it might be reasonable to not check the "suspiciousness" of this field inside the build method, and let to be checked in a future use case, that will be able to link the ExecParamRepo and the Transfer entity.

Reply:
After discussion, the use case ProcessParsedTransfer can be used to pass the values (retrieved from the ExecParamRepo) to the builder build method.

Reply:
See https://github.com/MobilityData/gtfs-validator/issues/246